packages/shared/src/components/FeedItemComponent.tsx

```diff
@@ -159,7 +159,63 @@ const getTags = ({
   };
 };
 
-export default function FeedItemComponent({
+export const withFeedLogExtraContext = (
+  WrappedComponent: typeof FeedItemComponent,
+): typeof FeedItemComponent => {
+  const WithFeedLogExtraContext = (
+    props: FeedItemComponentProps,
+  ): ReactElement => {
+    const { item } = props;
+
+    if (item?.type === FeedItemType.Post || isBoostedPostAd(item)) {
+      const post =
+        item.type === FeedItemType.Post ? item.post : item.ad.data?.post;
+      const ad = item.type === FeedItemType.Ad ? item.ad : undefined;
+
+      <ActivePostContextProvider post={post}>
+        <LogExtraContextProvider
```

Comment:
But we already wrapped everything in this right?
Why the differ of renders now?

Reply:
I mean in original, could you not just have done:
```
gen_id: ad?.generationId,
```
It would only add if exists anyway?